pip_check_reqs/common.py

```diff
@@ -77,41 +77,32 @@ def _add_module(self, modname: str, lineno: int) -> None:
             return
         path = None
         progress = []
-        modpath = last_modpath = None
+        modpath = None
         for modname_part in modname.split("."):
-            try:
-                _, modpath, _ = imp.find_module(modname_part, path)
-            except ImportError:
-                # the component specified at this point is not importable
-                # (is just an attribute of the module)
-                # *or* it's not actually installed, so we don't care either
-                break
-
-            # success! we found *something*
-            progress.append(modname_part)
+            find_spec_result = importlib.util.find_spec(
+                name=modname_part,
+                package=path,
+            )
 
-            # we might have previously seen a useful path though...
-            if modpath is None:
-                # the `sys` module will hit this code path, and `os` will on
-                # 3.11+.
-                # Possibly others will, but I've not discovered them.
-                modpath = last_modpath
+            if find_spec_result is None:
+                # The component specified at this point is not installed.
```

Comment:
IMO we should be warning here, but I guess it's a separate issue.

Reply:
Yes - I think the test case for that is `import foo` where `foo` does not exist.